test/test_tune.py

```diff
@@ -116,49 +116,102 @@ def test_autotune_match(self, open_db_mock: mock.Mock, close_db_mock: mock.Mock,
+        # re-run. This also populates the database with another set
+        tuning_2 = {'a': 3, 'b': 4}
+        self.autotune_mock.return_value = tuning_2
         context.device.driver_version = 'another version'
         context.device.platform_name = 'another platform'
         context.device.name = 'another device'
-        ret3 = self.autotune(context, 'xyz')
-        assert ret3 == another_tuning
+        ret = self.autotune(context, 'xyz')
+        assert ret == tuning_2
 
-        # Now test incremental fallback on device  [ driver version | platform | name ] when
-        # KATSDPSIGPROC_TUNE_MATCH is set to 'nearest'
+        # Now test incremental fallback on device  [ driver version | platform | name ]
+        # when KATSDPSIGPROC_TUNE_MATCH is set to 'nearest'
         with mock.patch('katsdpsigproc.tune.KATSDPSIGPROC_TUNE_MATCH', 'nearest'):
-            # Re-run with different configurations, and make sure it always uses cached values.
+            # autotuning re-runs if arg_parameters are different
+            tuning_4 = {'a': 7, 'b': 8}
+            self.autotune_mock.return_value = tuning_4
+            context.device.driver_version = 'mock version'
+            context.device.platform_name = 'mock platform'
+            context.device.name = 'mock device'
+            ret = self.autotune(context, 'zzz')
+            assert ret == tuning_4
+
+            # Re-run with different configurations making sure the cached values are returned if
+            # they exist in any configuration, and that autotuning does not run.
             self.autotune_mock.side_effect = RuntimeError
 
             context.device.driver_version = 'abc'
             context.device.platform_name = 'mock platform'
             context.device.name = 'mock device'
-            ret4 = self.autotune(context, 'xyz')
-            assert ret4 == tuning
+            ret = self.autotune(context, 'xyz')
+            assert ret == tuning_1
 
             context.device.driver_version = 'abc'
             context.device.platform_name = 'abc'
             context.device.name = 'another device'
-            ret5 = self.autotune(context, 'xyz')
-            assert ret5 == another_tuning
+            ret = self.autotune(context, 'xyz')
+            assert ret == tuning_2
 
-            # If there is no match in the database, any tuning will suffice
-            context.device.driver_version = 'abc'
-            context.device.platform_name = 'abc'
-            context.device.name = 'abc'
-            ret6 = self.autotune(context, 'xyz')
-            assert ret6 == tuning or ret6 == another_tuning
+            context.device.driver_version = 'mock version'
+            context.device.platform_name = 'mock platform'
+            context.device.name = 'mock device'
+            ret = self.autotune(context, 'xyz')
+            assert ret == tuning_1
+
+
+class TestAutotunerNoParam:
+    """Tests for class that implements autotune with no parameters
+    """
```

Comment:
For a one-line docstring, put the closing triple-quote on the same line as the text.